JabbR/Commands/GiphyCommand.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.IO;
+using System.Net;
+using System.Text;
+using JabbR.Models;
+using Microsoft.Ajax.Utilities;
+using Microsoft.AspNet.SignalR;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using RestSharp;
+using Http = JabbR.Infrastructure.Http;
+
+namespace JabbR.Commands
+{
+    [Command("giphy","Giphy_CommandInfo", "keyword1 keyword2 keywordX", "user")]
+    public class GiphyCommand : UserCommand
+    {
+        public override void Execute(CommandContext context, CallerContext callerContext, ChatUser callingUser, string[] args)
+        {
+            if (String.IsNullOrEmpty(callerContext.RoomName))
+            {
+                throw new HubException(LanguageResources.InvokeFromRoomRequired);
+            }
+
+            ChatRoom callingRoom = context.Repository.GetRoomByName(callerContext.RoomName);
+
+            string url = String.Format("http://api.giphy.com/v1/gifs/random?api_key={0}&tag=", "dc6zaTOxFJmzC");
+
+            //add keywords/tags to the url
+            int i = 0;
+            foreach (String keyword in args)
```

Comment:
Looks like [the API](https://github.com/Giphy/GiphyAPI#random-endpoint) doesn't support joining keywords in this way. it seems they need to be separated by a `space` character (which when encoded as [application/x-www-form-urlencoded](http://www.w3.org/TR/html401/interact/forms.html#h-17.13.4.1) is represents as `+`).

Also consider sanitising the input:

```C#
var tags = string.Join("+", args.Select(arg => Uri.EscapeDataString(arg)));
var url = string.Format("http://api.giphy.com/v1/gifs/random?api_key={0}&tag={1}", apiKey, tags);
```